### PR TITLE
/vg/ temporary fork stuff.

### PR DESCRIPTION
**Pull request by PJB3005**
Yeah I can't be bothered to list these changes out properly in individual PRs, commits should be semi-sane.

This was from the temporary repo we were using while I was trying to contact @Silvertorch5 via Reddit and we weren't sure whether he was alive right now.

I'm gonna go through the diffs shortly to fix anything Xamarin or VS fucked up with the project files.

### CODE REVIEW

File: SS14.Client.Services/SS14.Client.Services.csproj [hunk 1]
```diff
@@ -484,23 +486,13 @@
     <Compile Include="Utility\RobustSortedDrawing.cs">
       <SubType>Code</SubType>
     </Compile>
-    <EmbeddedResource Include="Input\KeyBindings.xsd">
```
Comment (PJB3005) [inline SS14.Client.Services/SS14.Client.Services.csproj]: Well this was probably important I guess.

**Comment by PJB3005** [comment]
~~\> Xamarin defaults to tabs but VS defaults to spaces.~~

~~I'm gonna punch somebody at Microsoft.~~

~~Probably the moron that decided EditorConfig isn't important enough for Xamarin.~~

EDIT: Ah you can store code formatting in the solution file that's handy.

File: SS14.Client/SpaceStation14.csproj [hunk 2]
```diff
@@ -27,6 +26,21 @@
     <FileUpgradeFlags>
     </FileUpgradeFlags>
     <TargetFrameworkProfile />
+    <PublishUrl>publish\</PublishUrl>
```
Comment (PJB3005) [inline SS14.Client/SpaceStation14.csproj:29]: Any idea what these changes are and whether they're important? Xamarin decided to do this.

File: SS14.Client/SpaceStation14.csproj [hunk 3]
```diff
@@ -260,11 +274,21 @@
       <DesignTime>True</DesignTime>
     </Compile>
   </ItemGroup>
+  <ItemGroup>
```
Comment (PJB3005) [inline SS14.Client/SpaceStation14.csproj]: Xamarin please.

**Comment by psykzz** [comment]
On merging can we squash the PR and just give a detailed commit message?

**Comment by PJB3005** [comment]
@psykzz sounds good I'll write one up in a sec.

**Comment by PJB3005** [comment]
```
* Removed SS14.Client/TextureVisualizer.cs, wasn't useful and broke builds on VS2017.
* Removed the bare-bones Lua scripting implementation as it was worthless and we probably aren't going to use Lua.
* Fixed tons of issues with GUI code so that the DebugConsole and ChatBox work. (Scrolling, textboxes, line length handling, focusing issues)
* Reworked handling of console and chat commands, except for server side client commands which aren't implemented yet.
* Added a .editorconfig and edited solution policies to fix formatting issues when Xamarin disagrees with VS.
* Made IoC and the server less working directory dependent and go relative to the directory containing the executable. Not done yet.
```

**Comment by ComicIronic** [comment]
Conflicts now.

**Comment by ComicIronic** [comment]
Any objection to me cherrypicking these out into individual PRs for better review? I'd particularly like to be able to look over the merits of removing all the Lua stuff thus far, because there might be some common interfaces worth keeping.

**Comment by PJB3005** [comment]
@ComicIronic nah no objections.

Though really there is nothing worth keeping about the Lua stuff.

**Comment by ZoldorfTheWizard** [comment]
Fix conflicts plz

**Comment by PJB3005** [comment]
Comic separated this into multiple PRs so I think we can close it.

**Pull request closed without merging**